packages/core/src/diff/field.ts

```diff
@@ -52,36 +52,41 @@ export function changesInField(
     }
   }
 
-  if (isNotEqual(oldField.type.toString(), newField.type.toString())) {
-    addChange(fieldTypeChanged(type, oldField, newField));
+  if (!isVoid(oldField) && isNotEqual(oldField!.type.toString(), newField.type.toString())) {
+    addChange(fieldTypeChanged(type, oldField!, newField));
   }
 
-  compareLists(oldField.args, newField.args, {
+  compareLists(oldField?.args ?? [], newField.args, {
     onAdded(arg) {
       addChange(fieldArgumentAdded(type, newField, arg));
     },
     onRemoved(arg) {
-      addChange(fieldArgumentRemoved(type, oldField, arg));
+      addChange(fieldArgumentRemoved(type, newField, arg));
     },
     onMutual(arg) {
-      changesInArgument(type, oldField, arg.oldVersion, arg.newVersion, addChange);
+      changesInArgument(type, newField, arg.oldVersion, arg.newVersion, addChange);
     },
   });
 
-  compareLists(oldField.astNode?.directives || [], newField.astNode?.directives || [], {
+  compareLists(oldField?.astNode?.directives || [], newField.astNode?.directives || [], {
     onAdded(directive) {
       addChange(
-        directiveUsageAdded(Kind.FIELD_DEFINITION, directive, {
-          parentType: type,
-          field: newField,
-        }),
+        directiveUsageAdded(
```

Comment:
Should this filter out the deprecated directive since that's captured by another change?